Subject: Enable Boot Complete CLI
diff --color -Naur deb-pkg-ostree-orig/src/ostree/ot-builtin-admin.c deb-pkg-ostree-working-copy/src/ostree/ot-builtin-admin.c
--- deb-pkg-ostree-orig/src/ostree/ot-builtin-admin.c	2023-09-06 16:08:48.152444851 +0200
+++ deb-pkg-ostree-working-copy/src/ostree/ot-builtin-admin.c	2023-09-11 11:31:17.884197376 +0200
@@ -39,7 +39,7 @@
     "Checkout revision REFSPEC as the new default deployment" },
   { "finalize-staged", OSTREE_BUILTIN_FLAG_NO_REPO | OSTREE_BUILTIN_FLAG_HIDDEN,
     ot_admin_builtin_finalize_staged, "Internal command to run at shutdown time" },
-  { "boot-complete", OSTREE_BUILTIN_FLAG_NO_REPO | OSTREE_BUILTIN_FLAG_HIDDEN,
+  { "boot-complete", OSTREE_BUILTIN_FLAG_NO_REPO,
     ot_admin_builtin_boot_complete, "Internal command to run at boot after an update was applied" },
   { "init-fs", OSTREE_BUILTIN_FLAG_NO_REPO, ot_admin_builtin_init_fs,
     "Initialize a root filesystem" },
