diff --color -Naur deb-pkg-ostree-orig/src/ostree/main.c deb-pkg-ostree-working-copy/src/ostree/main.c
--- deb-pkg-ostree-orig/src/ostree/main.c	2023-09-06 16:08:48.148445008 +0200
+++ deb-pkg-ostree-working-copy/src/ostree/main.c	2023-09-12 15:49:41.678766350 +0200
@@ -76,6 +76,10 @@
   { "static-delta", OSTREE_BUILTIN_FLAG_NONE, ostree_builtin_static_delta,
     "Static delta related commands" },
   { "summary", OSTREE_BUILTIN_FLAG_NONE, ostree_builtin_summary, "Manage summary metadata" },
+  { "hello-ostree",               // The name of the command
+   OSTREE_BUILTIN_FLAG_NO_REPO,  // Flag not to require the `--repo` argument, see "ot-main.h"
+   ostree_builtin_hello_ostree,  // Execution function for the command
+   "Print hello message" },      // Short description to appear when `ostree hello-ostree --help` is entered
   { NULL }
 };
 
diff --color -Naur deb-pkg-ostree-orig/src/ostree/ot-builtin-hello-ostree.c deb-pkg-ostree-working-copy/src/ostree/ot-builtin-hello-ostree.c
--- deb-pkg-ostree-orig/src/ostree/ot-builtin-hello-ostree.c	1970-01-01 01:00:00.000000000 +0100
+++ deb-pkg-ostree-working-copy/src/ostree/ot-builtin-hello-ostree.c	2023-09-12 15:48:37.618517530 +0200
@@ -0,0 +1,29 @@
+ #include "config.h"
+
+ #include "ot-main.h"
+ #include "ot-builtins.h"
+ #include "ostree.h"
+ #include "otutil.h"
+
+ // Structure for options such as ostree hello-ostree --option.
+ static GOptionEntry options[] = {
+   { NULL },
+ };
+
+ gboolean
+ ostree_builtin_hello_ostree (int argc, char **argv, OstreeCommandInvocation *invocation, GCancellable *cancellable, GError **error)
+ {
+   g_autoptr(OstreeRepo) repo = NULL;
+
+   // Creates new command context, ready to be parsed.
+   // Input to g_option_context_new shows when running ostree <command> --help
+   g_autoptr(GOptionContext) context = g_option_context_new ("");
+
+   // Parses the command context according to the ostree CLI.
+   if (!ostree_option_context_parse (context, options, &argc, &argv, invocation, &repo, cancellable, error))
+     return FALSE;
+
+   g_print("Hello OSTree!\n");
+
+   return TRUE;
+ }
diff --color -Naur deb-pkg-ostree-orig/src/ostree/ot-builtins.h deb-pkg-ostree-working-copy/src/ostree/ot-builtins.h
--- deb-pkg-ostree-orig/src/ostree/ot-builtins.h	2023-09-06 16:08:48.152444851 +0200
+++ deb-pkg-ostree-working-copy/src/ostree/ot-builtins.h	2023-09-12 15:50:54.245285667 +0200
@@ -61,6 +61,7 @@
 BUILTINPROTO (rev_parse);
 BUILTINPROTO (remote);
 BUILTINPROTO (write_refs);
+BUILTINPROTO (hello_ostree);
 
 #undef BUILTINPROTO
 
