diff --color -Naur deb-pkg-ostree-orig/src/libostree/ostree-sysroot-upgrader.c deb-pkg-ostree-working-copy/src/libostree/ostree-sysroot-upgrader.c
--- deb-pkg-ostree-orig/src/libostree/ostree-sysroot-upgrader.c	2023-09-06 16:08:48.148445008 +0200
+++ deb-pkg-ostree-working-copy/src/libostree/ostree-sysroot-upgrader.c	2023-09-20 11:30:26.197032024 +0200
@@ -562,6 +562,8 @@
         return FALSE;
     }
 
+  g_print ("\n>>>DBG: from_revision: %s\n", from_revision);
+  g_print (">>>DBG: new_revision: %s)\n\n", self->new_revision);
   if (g_strcmp0 (from_revision, self->new_revision) == 0)
     {
       *out_changed = FALSE;
