diff --color -Naur deb-pkg-ostree-orig/src/switchroot/ostree-prepare-root.c deb-pkg-ostree-working-copy/src/switchroot/ostree-prepare-root.c
--- deb-pkg-ostree-orig/src/switchroot/ostree-prepare-root.c	2023-09-06 16:08:48.152444851 +0200
+++ deb-pkg-ostree-working-copy/src/switchroot/ostree-prepare-root.c	2023-09-25 15:38:05.147081727 +0200
@@ -672,6 +672,10 @@
    * root under /sysroot/sysroot as systemd will be responsible for
    * moving /sysroot to /.
    */
+
+  if (mkdirat (AT_FDCWD, "sysroot", 0755) < 0)
+    g_print (">>>DBG>>: Failed to create %s", "sysroot");
+
   if (mount (root_mountpoint, "sysroot", NULL, MS_MOVE | MS_SILENT, NULL) < 0)
     err (EXIT_FAILURE, "failed to MS_MOVE '%s' to 'sysroot'", root_mountpoint);
 
